backend.native/compiletest/src/test/kotlin/org/jetbrains/kotlin/compiletest/LldbTests.kt

```diff
@@ -0,0 +1,56 @@
+import org.jetbrains.kotlin.compiletest.lldbTest
+import org.junit.Test
+
+class LldbTests {
+    //FIXME: the last one should be main.kt.5
+    @Test
+    fun `can step through code`() = lldbTest("""
+        fun main(args: Array<String>) {
+            var x = 1
+            var y = 2
+            var z = x + y
+            println(z)
+        }
+    """, """
+        > b main.kt:2
+        Breakpoint 1:
+
+        > r
+        Process [..] stopped
+        [..] at main.kt:2, [..] stop reason = breakpoint
+
+        > n
+        Process [..] stopped
+        [..] at main.kt:3, [..] stop reason = step over
+
+        > n
+        Process [..] stopped
+        [..] at main.kt:4, [..] stop reason = step over
+
+        > n
+        Process [..] stopped
+        [..] at main.kt:3, [..] stop reason = step over
+    """)
```

Comment:
Here's how inline tests look like:

![image](https://user-images.githubusercontent.com/1711539/31668988-627b144c-b35c-11e7-8624-93eda288dcd2.png)

They are big, but you actually see what you are testing, and it's always possible to split a huge file into smaller test suites :) 

I am not married to this approach though, specifying test fixtures as files is fine by me!